src/test/esm_exceptions.sol

```diff
@@ -0,0 +1,219 @@
+// SPDX-FileCopyrightText: © 2023 Dai Foundation <www.daifoundation.org>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+pragma solidity 0.8.16;
+
+contract EsmExceptions {
+
+    bytes32[] public exceptions;
+
+    constructor() {
+        exceptions = [           // Why?
+            bytes32("MCD_ESM"),  // Self-referential
+            "PROXY_DEPLOYER",    // Ecosystem tool
+            "WETH",              // External contract, not PP ward
+            "ETH",               // External contract, not PP ward
+            // Everything after this should be evaluated for security
```

Comment:
A note for future maintainers of this codebase, that this list would need to be carefully inspected to ensure auth on ESM is not required (e.g. the mudule is self-protected during ES via `vat.live` check or disabled by `END.cage`).

Reply:
Even if some module is protected, wouldn't it be better to always `rely` on the ESM contract?

Other than additional gas costs or maybe a faulty ESM code (which shouldn't happen anyway), I can't see a good reason to not do it.

Otherwise, we have yet another blacklist to be manually maintained in the future.

Reply:
https://forum.makerdao.com/t/overlooked-vectors-for-post-shutdown-governance-attacks-postmortem/20696

Reply:
I've read it before commenting. I believe you are referring to this part:

> Generally, most modules could be excluded from needing to authorize the ESM based on simple criteria like not having wards, lacking sensitive permissions, having embedded safety mechanisms, being external to the system, or not being upgradable.

And:

> Explicit justifications for “safe” contracts are not included here as these ought to be independently verified by whoever assumes responsibility for system safety post-PECU shutdown.

While I agree, I can't help but wonder if the cost of maintaining this blacklist of contracts that don't need to be checked would not be higher than simply having a rule to `rely` on the ESM everywhere the `PauseProxy` is `rely`ed.

Reply:
https://forum.makerdao.com/t/overlooked-vectors-for-post-shutdown-governance-attacks-postmortem/20696/2